packages/labs/context/src/lib/controllers/context-container.ts

```diff
@@ -0,0 +1,77 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {ContextCallback} from '../context-event.js';
+
+/**
+ * A disposer function
+ */
+type Disposer = () => void;
+
+/**
+ * A tuple of context callback, and its disposer
+ */
+type CallbackRecord<T> = [ContextCallback<T>, Disposer];
+
+/**
+ * A simple class which stores a value, and triggers registered callbacks when the
+ * value is changed via its setter.
+ *
+ * An implementor might use other observable patterns such as MobX or Redux to get
+ * behavior like this. But this is a pretty minimal approach that will likely work
+ * for a number of use cases.
+ */
+export class ContextContainer<T> {
```

Comment:
This file could be used outside of Lit.

Has this been discussed in the CG of the context community-protocol breakout notes?

Reply:
It's basically a "Container" or "Bottle" with list of listener for that. A bit like what vue-next does for dependency tracking.